packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -60,15 +66,25 @@ contract BridgeRouter is
     mapping(uint16 chainId => address routerAddress)
         public chainToRouterAddress;
 
+    /// @notice Reference to the cross-chain registry
+    ICrossChainRegistry public immutable registry;
+
     /*//////////////////////////////////////////////////////////////
                             CONSTRUCTOR
     //////////////////////////////////////////////////////////////*/
 
     /**
      * @notice Initializes the BridgeRouter contract
      * @param accessManager Address of the ProtocolAccessManager contract
+     * @param _registry Address of the CrossChainRegistry contract
      */
-    constructor(address accessManager) ProtocolAccessManaged(accessManager) {}
+    constructor(
+        address accessManager,
+        address _registry
+    ) ProtocolAccessManaged(accessManager) {
+        if (_registry == address(0)) revert InvalidParams();
+        registry = ICrossChainRegistry(_registry);
```

Comment:
`CrossChainConfigManaged` could contain helpers such as `isExecutor()` or simialr for easier registry integration.